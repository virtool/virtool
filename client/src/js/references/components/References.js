import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";

import OTUDetail from "../../otus/components/Detail/Detail";
import IndexDetail from "../../indexes/components/Detail";
import ReferenceList from "./List";
import ReferenceDetail from "./Detail/Detail";
import { LoadingPlaceholder } from "../../base";
import SourceTypes from "../../administration/components/General/SourceTypes";

export const ReferenceSettings = () => (
    <div className="settings-container">
        <h3 className="view-header">
            <strong>
                Settings
            </strong>
        </h3>
        <SourceTypes />
    </div>
);

const References = (props) => {
    if (props.settings === null) {
        return <LoadingPlaceholder />;
    }

    return (
        <div className="container">
            <Switch>
                <Route path="/refs" component={ReferenceList} exact />
                <Redirect from="/refs/settings/*" to="/refs/settings" />
                <Route path="/refs/settings" component={ReferenceSettings} />
                <Route path="/refs/:refId/otus/:otuId" component={OTUDetail} />
                <Route path="/refs/:refId/indexes/:indexId" component={IndexDetail} />
                <Route path="/refs/:refId" component={ReferenceDetail} />
            </Switch>
        </div>
    );
};

const mapStateToProps = (state) => ({
    settings: state.settings.data
});

export default connect(mapStateToProps)(References);
