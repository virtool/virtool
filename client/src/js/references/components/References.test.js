import References, { ReferenceSettings } from "./References";

describe("<References />", () => {
    let initialState;
    let store;
    let wrapper;

    it("renders correctly with settings data", () => {
        initialState = {
            settings: {
                data: { foo: true, bar: false }
            }
        };
        store = mockStore(initialState);
        wrapper = shallow(<References store={store} />).dive();
        expect(wrapper).toMatchSnapshot();
    });

    it("renders <LoadingPlaceholder /> when no settings data available", () => {
        initialState = { settings: { data: null } };
        store = mockStore(initialState);
        wrapper = shallow(<References store={store} />).dive();
        expect(wrapper).toMatchSnapshot();
    });

    it("renders <ReferenceSettings /> correctly", () => {
        wrapper = shallow(<ReferenceSettings />);
        expect(wrapper).toMatchSnapshot();
    });

});
