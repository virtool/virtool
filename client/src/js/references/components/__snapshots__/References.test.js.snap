// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<References /> renders <LoadingPlaceholder /> when no settings data available 1`] = `<LoadingPlaceholder />`;

exports[`<References /> renders <ReferenceSettings /> correctly 1`] = `
<div
  className="settings-container"
>
  <h3
    className="view-header"
  >
    <strong>
      Settings
    </strong>
  </h3>
  <Connect(SourceTypes) />
</div>
`;

exports[`<References /> renders correctly with settings data 1`] = `
<div
  className="container"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/refs"
    />
    <Redirect
      from="/refs/settings/*"
      push={false}
      to="/refs/settings"
    />
    <Route
      component={[Function]}
      path="/refs/settings"
    />
    <Route
      component={[Function]}
      path="/refs/:refId/otus/:otuId"
    />
    <Route
      component={[Function]}
      path="/refs/:refId/indexes/:indexId"
    />
    <Route
      component={[Function]}
      path="/refs/:refId"
    />
  </Switch>
</div>
`;
